build/build-in-docker

```diff
@@ -24,30 +24,39 @@ set -e
 SCRIPTDIR=$(cd $(dirname $0); pwd)
 
 LOCAL_MAVEN_REPO=${LOCAL_MAVEN_REPO:-"$HOME/.m2/repository"}
-CUDF_USE_PER_THREAD_DEFAULT_STREAM=${CUDF_USE_PER_THREAD_DEFAULT_STREAM:-ON}
 USE_GDS=${USE_GDS:-ON}
 export CMAKE_GENERATOR=${CMAKE_GENERATOR:-"Ninja"}
+export DOCKER_BUILD_EXTRA_ARGS="--platform=linux/amd64 --build-arg CMAKE_ARCH=x86_64"
+CUDA_VER=${CUDA_VER:-cuda11}
+USE_SANITIZER=${USE_SANITIZER:-ON}
+BUILD_FAULTINJ=${BUILD_FAULTINJ:-ON}
 
 if (( $# == 0 )); then
   echo "Usage: $0 <Maven build arguments>"
   exit 1
 fi
 
-_CUDF_CLEAN_SKIP=""
-# if ccache is enabled and libcudf.clean.skip not provided
-# by the user remove the cpp build directory
-#
-if [[ "$CCACHE_DISABLE" != "1" ]]; then
-  if [[ ! "$*" =~ " -Dlibcudf.clean.skip=" ]]; then
-    # Don't skip clean if ccache is enabled
-    # unless the user overrides
-    _CUDF_CLEAN_SKIP="-Dlibcudf.clean.skip=false"
-  fi
+case $(uname -m) in
+  x86_64|amd64)
+    arch=amd64;;
+  aarch64|arm64)
+    arch=arm64;;
+  *)
+    echo "Unsupported CPU architecture"; exit 1;;
+esac
+
+# Set env for arm64 build
+if [ "$arch" == "arm64" ]; then
+  profiles="${profiles},arm64"
+  USE_GDS="OFF"
```

Comment:
Curious why we don't build GDS on arm64?  It used to be separate but now is part of the CUDA toolkit.  Is it not part of the arm64 CUDA toolkit?

Reply:
> Curious why we don't build GDS on arm64? It used to be separate but now is part of the CUDA toolkit. Is it not part of the arm64 CUDA toolkit?

Yes, GDS cuFiles lib is not in the arm64 CUDA toolkit

[INFO]      [exec]   Could NOT find cuFile (missing: cuFile_LIBRARY cuFileRDMA_LIBRARY
[INFO]      [exec]   cuFile_INCLUDE_DIR)

Reply:
We need to fix the tests on arm64, not hack the build script to run the sanitizer to hide the problem.  We will not be running the sanitizer in production, so letting this slip through by hacking the build script is not what we want.